CONTRIBUTING.md

```diff
@@ -0,0 +1,63 @@
+# Introduction
+
+Thank you for contributing to DCS-gRPC. Before submitting your PR please read through the following tenets we aim
+to follow on this project 
+
+# Tenets
+
+# KISS (Keep It Simple, Stupid) for clients
+
+DCS-gRPC clients can be written in a wide variety of languages, each with their own idiosyncracies. Therefore we will
+focus on keeping the gRPC interface as simple as possible even if this means some extra verbosity or complexity inside
+the DCS-gRPC server.
+
+# Maintain consistency across the DCS-gRPC APIs
+
+Try and maintain consistency in call patterns and field names in the DCS-gRPC API. This may mean breaking from the 
+conventions in the underlying ED APIs (See the next Tenet)
+
+# Follow ED API conventions by default but do not be slaves to them.
+
+We will follow the ED API conventions by default but this is a guide rather than a rule. Renaming fields to make more
+sense is fine for example.
+
+# Document the gRPC interface
+
+Add documentation to the gRPC .proto files using the Proto-Gen-Doc format detailed at https://github.com/pseudomuto/protoc-gen-doc#writing-documentation
+
+# Follow git commit message best practices
+
+Do not create 1 line commit messages for anything but the most trivial of commits. Follow this template for 
+
+```plain
+Capitalized, short (50 chars or less) summary
+
+More detailed explanatory text, if necessary.  Wrap it to about 72
+characters or so.  In some contexts, the first line is treated as the
+subject of an email and the rest of the text as the body.  The blank
+line separating the summary from the body is critical (unless you omit
+the body entirely); tools like rebase can get confused if you run the
+two together.
+
+Try to pre-emptively answer any foreseeable "Why?" questions a reader
+may have. There is no size limit on commit messages.
+
+Write your commit message in the imperative: "Fix bug" and not "Fixed bug"
+or "Fixes bug."  This convention matches up with commit messages generated
+by commands like git merge and git revert.
+
+Further paragraphs come after blank lines.
+
+- Bullet points are okay, too
+
+- Typically a hyphen or asterisk is used for the bullet, followed by a
+  single space, with blank lines in between, but conventions vary here
+
+- Use a hanging indent
+
+If you use an issue tracker, add a reference(s) to them at the bottom,
+like so:
+
+Resolves: #123
+```
```

Comment:
New heading - Git history

Points to cover
* Squash commits to create a cognitive timeline
* Upon request for changes / reviews, rewrite history and rebase

Tips:
* If needing to pull upstream changes, rebase the branch on top of `main`. This would avoid the multi-parent squash balancing act.

Reply:
Added info about communication channels, squashing and linters.